tidy3d/plugins/waveguide/rectangular_dielectric.py

```diff
@@ -89,17 +92,19 @@ class RectangularDielectric(Tidy3dBaseModel):
         units=MICROMETER,
     )
 
-    clad_thickness: Size1D = pydantic.Field(
+    clad_thickness: Union[Size1D, ArrayFloat1D] = pydantic.Field(
```

Comment:
might need to validate that the array is all non-negative as I'm not sure it will check this if an array is supplied.

Reply:
Not sure about performance, and if it becomes annoying in the code, but could also just make it a `Tuple[Size1D, ...]`

Reply:
I've added a few tests for this and it seems to work as intended.